### PR TITLE
Add SavedAsPointer and it's mutable part as trait

### PR DESCRIPTION
**Pull request by kloenk**
This will be used for NetDevice and NlAttr wrappers in #439

### CODE REVIEW

File: rust/kernel/types.rs [hunk 1]
```diff
@@ -247,3 +247,52 @@ impl<T: FnOnce()> Drop for ScopeGuard<T> {
         }
     }
 }
+
+/// Trait for wrappers which are holding a direct pointer that they are getting from a C side function.
+pub trait SavedAsPointer {
```
Comment (wedsonaf) [inline rust/kernel/types.rs:252]: Is there a reason why you can't use `PointerWrapper`? We should unify all these pointer wrappers.

Comment (kloenk) [reply rust/kernel/types.rs:252]: My Understanding of PointerWrapper is that it is designed for rust types which can be given to a C interface. The trait I am implementing is designed to hold a pointer which got allocated via a C api.

Comment (kloenk) [reply rust/kernel/types.rs:252]: @wedsonaf is my understanding of `PointerWrapper` correct? if not, should I withdraw this pr?

**Comment by kloenk** [comment]
This is similar to #557 but that does not implement the get_internal functions. closing it, as it got stall

**Pull request closed without merging**